snakefmt/parser/parser.py

```diff
@@ -59,6 +59,14 @@ def __init__(self, snakefile: TokenIterator):
                 break
 
             keyword = status.token.string
+
+            if status.token.type == tokenize.COMMENT:
+                # check if inline or standalone
+                is_standalone_comment = comment_start(status.token.line)
+                if is_standalone_comment:
+                    comment_indent = int(status.token.start[1] / len(TAB))
+                    self.buffer += TAB * comment_indent
```

Comment:
I'm a little wary of this, as instinctively I remember that all comments are treated/formatted in `formatter.py` and they should get properly indented there (of course, they don't really as we are having these issues). can you see a reason that indenting comments here could mess up with their formatting in `formatter.py`? If not, happy to do this.

Either way, I think we should use the logic in https://github.com/snakemake/snakefmt/blob/d233dfa958479f8ee3b4a341f6ccc3fb8ae9ab78/snakefmt/parser/parser.py#L26-L33

to, whenever we reach a COMMENT token, get the next token (t =next(self.snakefile)), put it back in the iterator (self.snakefile.denext(t)), and infer the indentation using that token (for e.g. if its a DEDENT) and `status.indent`. Because the input file may not be indented using `TAB`. It could be indented using, say, two spaces per indent, invalidating the line 66 arithmetic.

Reply:
Yeah, totally understand your wariness. 

It's so hard to try and retain all the reasons for some of this in my head as I was walking through the failing test line by line. 

I seem to remember part of the problem was that tokenize basically "ignores" comments. I found sometimes it didn't even emit a COMMENT token. And I think COMMENTS and DEDENTs together were somehow causing some issues with our indent arithmetic.

I'll update the tests with the new examples and try rolling back my changes and use your suggestion to see if it helps.